packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -13,6 +13,36 @@ import { viewNames } from '../../comments/CommentsViews';
 
 export const MAX_COLUMN_WIDTH = 720
 
+const POST_DESCRIPTION_EXCLUSIONS: RegExp[] = [/crossposted/i, /epistemic status/i];
+
+/** Get a og:description-appropriate description for a post */
+export const getPostDescription = (post: PostsWithNavigation | PostsWithNavigationAndRevision) => {
+  if (post.contents?.plaintextDescription) {
+    // concatenate the first few paragraphs together up to some reasonable length
+    const firstFewPars = post.contents.plaintextDescription
+      // paragraphs in the plaintext description are separated by double-newlines
+      .split(/\n\n/)
+      // get rid of bullshit opening text ('epistemic status' or 'crossposted from' etc)
+      .filter((par) => !POST_DESCRIPTION_EXCLUSIONS.some((re) => re.test(par)))
+      // concatenate paragraphs together with a delimiter, until they reach an
+      // acceptable length (target is 100-200 characters)
+      // this will return a longer description if one of the first couple of
+      // paragraphs is longer than 200
```

Comment:
You don't think it's worth a hard stop at 200 chars?

Reply:
Aw, let's try it